scripts/transition-staging-repository-state.sh

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env bash
+#
+# Copyright Kroxylicious Authors.
+#
+# Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+#
+
+set -euo pipefail
+
+usage() {
+      1>&2 cat << EOF
+usage: $0 <-a|-s (close|drop|release)|-h>
+ -a asserts that no staging repos exist
+ -s desired state to move the staging repository (close|drop|release)
+ -h this help message
+EOF
+exit 1
+}
+
+PLUGIN=org.sonatype.plugins:nexus-staging-maven-plugin:1.6.13
+MVN_ARGS=("--batch-mode" "--no-transfer-progress" "-DnexusUrl=https://s01.oss.sonatype.org/" "-DserverId=ossrh")
+
+ASSERT_NO_STAGING_REPOS="false"
+STATE=""
+while getopts ":s:ah" opt; do
+  case $opt in
+    a) ASSERT_NO_STAGING_REPOS="true"
+    ;;
+    s) STATE="${OPTARG}"
+    ;;
+    h) usage
+    ;;
+    \?) echo "Invalid option -${OPTARG}" >&2
+    exit 1
+    ;;
+  esac
+done
+
+# TODO: refactor to use the REST API https://oss.sonatype.org/nexus-staging-plugin/default/docs/index.html
```

Comment:
```suggestion
# TODO: refactor to use the REST API https://oss.sonatype.org/nexus-staging-plugin/default/docs/index.html to avoid the awkish hell
```